### PR TITLE
per tenant gate for structured metadata feature

### PR DESCRIPTION
**Pull request by owen-d**
Adds a per tenant configuration to block access to the upcoming structured-metadata push API extension.

### CODE REVIEW

File: docs/sources/configure/_index.md [hunk 1]
```diff
@@ -2701,6 +2701,10 @@ shard_streams:
 # the deprecated -replication-factor for backwards compatibility reasons.
 # CLI flag: -index-gateway.shard-size
 [index_gateway_shard_size: <int> | default = 0]
+
+# Allow user to send structured metadata (non-indexed labels) in push payload.
+# CLI flag: -validation.structured-metadata
+[structured_metadata: <boolean> | default = false]
```
Comment (stevendungan) [inline docs/sources/configure/_index.md:2707]: @owen-d do you think the naming convention will lead to any confusion down the road, and allow_structured_metadata is cleaner? or is this a non-concern 3 months down the road?

File: docs/sources/send-data/promtail/stages/non_indexed_labels.md [hunk 2]
```diff
@@ -5,12 +5,16 @@ description: non_indexed_labels stage
...
 {{% admonition type="warning" %}}
-Non-indexed labels is an experimental feature and is subject to change in future releases of Grafana Loki.
+Non-indexed labels is an experimental feature and is subject to change in future releases of Grafana Loki. 
 {{% /admonition %}}
 
 The labels stage is an action stage that takes data from the extracted map and
 modifies the [non-indexed labels]({{< relref "../../../get-started/labels/non-indexed-labels" >}}) set that is sent to Loki with the log entry.
 
+{{% admonition type="warning" %}}
+Non-indexed labels will be rejected by loki unless the `structured_metadata` per tenant configuration (in the `limits_config`) is enabled.
```
Comment (JStickler) [inline docs/sources/send-data/promtail/stages/non_indexed_labels.md:15]:
```suggestion
Non-indexed labels will be rejected by Loki unless you enable the `structured_metadata` per tenant configuration (in the `limits_config`).
```

**Review by JStickler — APPROVED** [review]
[docs team]  One small suggestions, otherwise LGTM.

File: docs/sources/configure/_index.md (hunk 1 above)
Comment (owen-d) [reply docs/sources/configure/_index.md:2707]: good point @stevendungan -- i'll change it

File: pkg/validation/validate.go [hunk 3]
```diff
@@ -58,8 +58,10 @@ const (
 	LabelValueTooLong         = "label_value_too_long"
 	LabelValueTooLongErrorMsg = "stream '%s' has label value too long: '%s'"
 	// DuplicateLabelNames is a reason for discarding a log line which has duplicate label names
-	DuplicateLabelNames         = "duplicate_label_names"
-	DuplicateLabelNamesErrorMsg = "stream '%s' has duplicate label name: '%s'"
+	DuplicateLabelNames                  = "duplicate_label_names"
+	DuplicateLabelNamesErrorMsg          = "stream '%s' has duplicate label name: '%s'"
+	DisallowedStructuredMetadata         = "disallowed_structured_metadata"
+	DisallowedStructuredMetadataErrorMsg = "stream '%s' has includes structured metadata, but this feature is disallowed. Please see `limits_config.structured_metadata` or contact your Loki administrator to enable it."
```
Comment (salvacorts) [inline pkg/validation/validate.go:64]:
```suggestion
	DisallowedStructuredMetadataErrorMsg = "stream '%s' includes structured metadata, but this feature is disallowed. Please see `limits_config.structured_metadata` or contact your Loki administrator to enable it."
```

**Review by salvacorts — APPROVED** [review]
LGTM. Just added a suggestion for a typo.

File: docs/sources/send-data/promtail/stages/non_indexed_labels.md [hunk 4]
```diff
@@ -5,12 +5,16 @@ description: non_indexed_labels stage
 # non_indexed_labels
 
 {{% admonition type="warning" %}}
-Non-indexed labels is an experimental feature and is subject to change in future releases of Grafana Loki.
+Non-indexed labels is an experimental feature and is subject to change in future releases of Grafana Loki.
```
Comment (vlad-diachenko) [inline docs/sources/send-data/promtail/stages/non_indexed_labels.md:8]: please pull the main, I have renamed this file

Comment (salvacorts) [reply docs/sources/send-data/promtail/stages/non_indexed_labels.md]: Done

**Pull request merged**